### PR TITLE
Explain how to let the system pick a server port.

### PR DESCRIPTION
**Pull request by erikcorry**
Add comments and example on how to specify port zero and let the system pick a free server port, then server HTTP on that port.

This closes issue #91.

### CODE REVIEW

File: src/server.toit [hunk 1]
```diff
@@ -47,10 +47,24 @@ class Server:
     root_certificates_ = root_certificates
     if max_tasks > 1: semaphore_ = monitor.Semaphore --count=max_tasks
 
+  /**
+  Sets up an HTTP server on the given interface and port.
+  Use $(listen server_socket handler) if you want to let the system
```
Comment (erikcorry) [inline src/server.toit:52]: Is this the right syntax?

Comment (floitsch) [reply src/server.toit:52]: yes.

File: src/server.toit [hunk 2]
```diff
@@ -47,10 +47,24 @@ class Server:
...
     server_socket := interface.tcp_listen port
     listen server_socket handler
 
+  /**
+  Sets up an HTTP server on the given TCP server socket.
+  This variant of server_socket gives you more control over the socket,
+    eg. to let the system pick a free port.  See examples/server.toit.
+  The handler is called for each incoming request with two arguments:
+    The $Request and a $ResponseWriter.
+  */
```
Comment (floitsch) [inline src/server.toit:67]: Maybe add a full example?
~~~suggestion

  # Examples
  ```
  import http
  import net
  main:
    network := net.open
    // Listen on a free port.
    tcp_socket := network.tcp_listen 0
    print "Server on http://localhost:$tcp_socket.local_address.port/"
    server := http.Server
    server.listen tcp_socket:: | request/http.Request writer/http.ResponseWriter |
      if request.path == "/":
        writer.headers.set "Content-Type" "text/html"
        writer.write "<html><body>hello world</body></html>
      writer.close
  ```
  */
~~~

**Review by floitsch — APPROVED** [review]
LGTM.

**Pull request merged**